fragments/master.sh

```diff
@@ -118,68 +164,38 @@ start_k8s(){
             '{ "Network": "10.1.0.0/16", "Backend": {"Type": "vxlan"}}'
 
     # iface may change to a private network interface, eth0 is for default
-    flannelCID=$(docker -H unix:///var/run/docker-bootstrap.sock run \
-        --restart=on-failure \
+    docker run \
+        --restart=${RESTART_POLICY} \
         -d \
         --net=host \
         --privileged \
         -v /dev/net:/dev/net \
+        -v ${FLANNEL_SUBNET_TMPDIR}:/run/flannel \
         ${FLANNEL_IMAGE} \
         /opt/bin/flanneld \
             --ip-masq="${FLANNEL_IPMASQ}" \
-            --iface="${FLANNEL_IFACE}")
-
-    sleep 8
-
-    # Copy flannel env out and source it on the host
-    docker -H unix:///var/run/docker-bootstrap.sock \
-        cp ${flannelCID}:/run/flannel/subnet.env .
-    source subnet.env
-
-    # Configure docker net settings, then restart it
-    case "${lsb_dist}" in
-        amzn)
-            DOCKER_CONF="/etc/sysconfig/docker"
-            echo "OPTIONS=\"\$OPTIONS --mtu=${FLANNEL_MTU} --bip=${FLANNEL_SUBNET}\"" | tee -a ${DOCKER_CONF}
-            ifconfig docker0 down
-            yum -y -q install bridge-utils && brctl delbr docker0 && service docker restart
-            ;;
-        centos)
-            DOCKER_CONF="/usr/lib/systemd/system/docker.service"
-            sed -i "/^ExecStart=/ s~$~ --mtu=${FLANNEL_MTU} --bip=${FLANNEL_SUBNET}~" ${DOCKER_CONF}
-            sed -i.bak 's/^\(MountFlags=\).*/\1shared/' ${DOCKER_CONF}
-            systemctl daemon-reload
-            if ! command_exists ifconfig; then
-                yum -y -q install net-tools
-            fi
-            ifconfig docker0 down
-            yum -y -q install bridge-utils && brctl delbr docker0 && systemctl restart docker
-            ;;
-        ubuntu|debian)
-            DOCKER_CONF="/etc/default/docker"
-            echo "DOCKER_OPTS=\"\$DOCKER_OPTS --mtu=${FLANNEL_MTU} --bip=${FLANNEL_SUBNET}\"" | tee -a ${DOCKER_CONF}
-            ifconfig docker0 down
-            apt-get install bridge-utils
-            brctl delbr docker0
-            service docker stop
-            while [ `ps aux | grep /usr/bin/docker | grep -v grep | wc -l` -gt 0 ]; do
-                echo "Waiting for docker to terminate"
-                sleep 1
-            done
-            service docker start
-            ;;
-        *)
-            echo "Unsupported operations system ${lsb_dist}"
-            exit 1
-            ;;
-    esac
+            --iface="${FLANNEL_IFACE}"
+
+	# Wait for the flannel subnet.env file to be created instead of a timeout. This is faster and more reliable
+	local SECONDS=0
+	while [[ ! -f ${FLANNEL_SUBNET_TMPDIR}/subnet.env ]]; do
+		if [[ ${SECONDS} == ${TIMEOUT_FOR_SERVICES} ]]; then
+			echo "flannel failed to start. Exiting..."
+			exit
+		fi
+		sleep 1
+	done
+
+    # The subnet.env is used for CNI plugin to generate network configuration file.
+    mkdir -p /var/run/flannel
+    cp ${FLANNEL_SUBNET_TMPDIR}/subnet.env /var/run/flannel/subnet.env
```

Comment:
this copy does not overwrite. It also does not work after reboot - because script is not executed.

We must keep default location /var/run/flannel/subnet.env.